From 1d77790120d4f1a3a79b6ce5d220f407165992fa Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:15 -0400
Subject: [powerpc] Export flush_icache_range

Message-id: <1410545655-205645-447-git-send-email-dzickus@redhat.com>
Patchwork-id: 94561
O-Subject: [RHEL7 PATCH 446/626] powerpc: Export flush_icache_range
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 28ea3c7529388c1e4177336a0e655c6c7272fd1c
Author: Jeff Mahoney <jeffm@suse.com>
Date:   Sun Apr 27 18:10:43 2014 -0400

    powerpc: Export flush_icache_range

    Commit aac416fc38c (lkdtm: flush icache and report actions) calls
    flush_icache_range from a module. It's exported on most architectures
    that implement it, but not on powerpc. This patch exports it to fix
    the module link failure.

    Signed-off-by: Jeff Mahoney <jeffm@suse.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/ppc_ksyms.c b/arch/powerpc/kernel/ppc_ksyms.c
index eaaca24..7f20621 100644
--- a/arch/powerpc/kernel/ppc_ksyms.c
+++ b/arch/powerpc/kernel/ppc_ksyms.c
@@ -118,6 +118,7 @@ EXPORT_SYMBOL(giveup_spe);
 EXPORT_SYMBOL(flush_instruction_cache);
 #endif
 EXPORT_SYMBOL(flush_dcache_range);
+EXPORT_SYMBOL(flush_icache_range);
 
 #ifdef CONFIG_SMP
 #ifdef CONFIG_PPC32
-- 
1.7.1